src/test/script_tapscript_tests.cpp

```diff
@@ -0,0 +1,1546 @@
+                                                  MissingDataBehavior::FAIL),
+                            "SIGHASH_SINGLE with in_pos(2) > #tx_to==1 is expected to fail");
+    }
+}
+
+BOOST_AUTO_TEST_CASE(signature_hash_schnorr_all_success_paths)
+{
+    // Our approach here will be to follow BIP-341's signature algorithm (with
+    // the BIP-342 extension) doing two things at once:
+    //   1) We'll set up the input arguments to `SignatureHashSchnorr` function
+    //      being tested, _and_
+    //   2) we'll _compute the hash of those fields ourselves_ exaxctly as
+    //      it is described in BIP-341 and BIP-342.
+    // Then we can compare the two.  We'll do this in a data-driven way for each
+    // of the different scenarios that the algorithm supports.
+    //
+    // In this way this test achieves 100% _path_ coverage of `SignatureHashSchnorr`
+    // (not just 100% _branch_ coverage).
+    // - Sadly, this isn't shown in the `lcov` reports.  There are still a few
+    //   red `-` marks left.  This is because:
+    //   1. `lcov` wasn't designed to handle death tests.
+    //   2. ??? Some other unknown reasons, possibly due to the instrumentation,
+    //      possibly due to `lcov` limitations.  You can see by the test output
+    //      (`-log_level=all`) or within a debugger that in fact _all_ branches
+    //      are taken when executing all the tests in this file.
+
+    // Here we define, and then generate, all combinations of the alternatives
+    // for the parameters that vary the signature combination algorithm
+
+    const vn_sequence<SigVersion> SigVersion_alternatives{
+        {SigVersion::TAPROOT, "TAPROOT"sv},
+        {SigVersion::TAPSCRIPT, "TAPSCRIPT"sv}
+    };
+
+    const vn_sequence<uint32_t> hash_type_output_alternatives{
+        {SIGHASH_DEFAULT, "SIGHASH_DEFAULT"sv},
+        {SIGHASH_ALL, "SIGHASH_ALL"sv},
+        {SIGHASH_NONE, "SIGHASH_NONE"sv},
+        {SIGHASH_SINGLE, "SIGHASH_SINGLE"sv}
+    };
+
+    const vn_sequence<uint32_t> hash_type_input_alternatives{
+        {0, "N/A"sv},
+        {SIGHASH_ANYONECANPAY, "SIGHASH_ANYONECANPAY"sv}
+    };
+
+    const vn_sequence<uint8_t> annex_alternatives{
+        {0, "no annex"sv},
+        {1, "annex present"sv}
+    };
+
+    const vn_sequence<bool> output_hash_alternatives{
+        {false, "output hash missing"sv},
+        {true, "output hash provided"sv}
+    };
+
+    {
+        const int nAlternatives = SigVersion_alternatives.size()
+                                  * hash_type_output_alternatives.size()
+                                  * hash_type_input_alternatives.size()
+                                  * annex_alternatives.size()
+                                  * output_hash_alternatives.size()
+                                  - 8 /* exclude SIGHASH_DEFAULT w/ SISHASH_ANYONECANPAY */;
+
+        BOOST_TEST_MESSAGE("Running " << nAlternatives << "scenarios");
```

Comment:
Noticed this in the test output.
```suggestion
        BOOST_TEST_MESSAGE("Running " << nAlternatives << " scenarios");
```

Reply:
Fixed.